server/auth.js

```diff
@@ -14,26 +15,71 @@ const dayjs = require("dayjs");
  * @returns {Promise<(Bean|null)>} User or null if login failed
  */
 exports.login = async function (username, password) {
+
     if (typeof username !== "string" || typeof password !== "string") {
         return null;
     }
-
-    let user = await R.findOne("user", " username = ? AND active = 1 ", [
-        username,
-    ]);
-
-    if (user && passwordHash.verify(password, user.password)) {
-        // Upgrade the hash to bcrypt
-        if (passwordHash.needRehash(user.password)) {
-            await R.exec("UPDATE `user` SET password = ? WHERE id = ? ", [
-                passwordHash.generate(password),
-                user.id,
-            ]);
+    if (process.env.AUTH_METHOD === "LDAP") {
+        let ldapAuthSuccess = false;
+        await new Promise((resolve, reject) => {
+            const client = ldap.createClient({
+                url: [ process.env.LDAP_ADDRESS ]
+            });
+            client.on("connectError", (err) => {
+                log.error("auth", "connecting ldap server failed");
+                resolve();
```

Comment:
please refactor this to reject/resolve with the value of `ldapAuthSuccess`.
This way, **future PRs** can't mess with the logic and introduce potential security flaws. (this logic is fine, but lets not leave a footgun lying around ^^)